### ISSUE TITLE
Optionally disable warning when concatenating dataframes with unknown divisions

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask#6175]
In https://github.com/dask/dask/blob/2d14075004fbca3ad7e5907f5f58ab8b05cdbdc2/dask/dataframe/multi.py#L1093-L1098 we (rightfully) warn when concatenating dask series / dataframes with unknown divisions

```python
import numpy as np
import dask.array as da
import dask.dataframe as dd

a = da.arange(12)
a._chunks = ((np.nan,),)

a1 = dd.from_dask_array(a)
a2 = dd.from_dask_array(a)
dd.concat([a1, a2], axis=1)
```

As the warning says, this is *generally* unsafe. But sometimes the calling function has a bit more context and knows that it's safe. In this case, it'd be nice to pass

```python
dd.concat([a1, a2], axis=1, ignore_unknown_divisions=True)
```

To have Dask not emit the warning.

**Comment by mrocklin** [comment]
That seems sensible to me

**Comment by 3cham** [comment]
Hi, I've created a PR for it here: https://github.com/dask/dask/pull/6210

**Comment by TomAugspurger** [comment]
Closed by #6210
